storage/innobase/xtrabackup/utils/percona-xtrabackup-8.0_builder.sh

```diff
@@ -0,0 +1,585 @@
+    get_tar "source_tarball"
+    rm -f *.dsc *.orig.tar.gz *.debian.tar* *.changes
+
+    TARFILE=$(basename $(find . -name 'percona-xtrabackup-*.tar.gz' | sort | tail -n1))
+    NAME=$(echo ${TARFILE}| awk -F '-' '{print $1"-"$2}')
+    VERSION=$(echo ${TARFILE%.tar.gz} | awk -F '-' '{print $3}')
+    SHORTVER=$(echo ${VERSION} | awk -F '.' '{print $1"."$2}')
+
+    echo "DEB_RELEASE=${DEB_RELEASE}" >> ${CURDIR}/percona-xtrabackup-8.0.properties
+
+    NEWTAR=${NAME}-80_${VERSION}.orig.tar.gz
+    mv ${TARFILE} ${NEWTAR}
+
+    tar xzf ${NEWTAR}
+    cd percona-xtrabackup-$VERSION
+    cp -av storage/innobase/xtrabackup/utils/debian .
+    dch -D unstable --force-distribution -v "${VERSION}-${DEB_RELEASE}" "Update to new upstream release Percona XtraBackup ${VERSION}"
+    dpkg-buildpackage -S
+
+    cd ${WORKDIR}
+
+    mkdir -p $WORKDIR/source_deb
+    mkdir -p $CURDIR/source_deb
+
+    cp *.dsc $WORKDIR/source_deb
+    cp *.orig.tar.gz $WORKDIR/source_deb
+    cp *.debian.tar.* $WORKDIR/source_deb
+    cp *.changes $WORKDIR/source_deb
+    cp *.dsc $CURDIR/source_deb
+    cp *.orig.tar.gz $CURDIR/source_deb
+    cp *.debian.tar.* $CURDIR/source_deb
+    cp *.changes $CURDIR/source_deb
+    
+}
+
+build_deb(){
+    if [ $DEB = 0 ]
+    then
+        echo "source deb package will not be created"
+        return;
+    fi
+    if [ "x$OS" = "xrmp" ]
+    then
+        echo "It is not possible to build source deb here"
+        exit 1
+    fi
+    for file in 'dsc' 'orig.tar.gz' 'debian.tar.*' 'changes'
+    do
+        get_deb_sources $file
+    done
+    cd $WORKDIR
+
+    echo "DEB_RELEASE=${DEB_RELEASE}" >> ${CURDIR}/percona-xtrabackup-8.0.properties
+    echo "DEBIAN_VERSION=${OS_NAME}" >> ${CURDIR}/percona-xtrabackup-8.0.properties
+    DSC=$(basename $(find . -name '*.dsc' | sort | tail -n 1))
+    DIRNAME=$(echo $DSC | sed -e 's:_:-:g' | awk -F'-' '{print $1"-"$2"-"$3"-"$4}')
+    VERSION=$(echo $DSC | sed -e 's:_:-:g' | awk -F'-' '{print $4}')
+    ARCH=$(uname -m)
+
+    dpkg-source -x $DSC
+    cd $DIRNAME
+    dch -m -D "$OS_NAME" --force-distribution -v "$VERSION-$DEB_RELEASE.$OS_NAME" 'Update distribution'
+    dpkg-buildpackage -rfakeroot -uc -us -b
+
+    cd ${WORKDIR}
+    mkdir -p $CURDIR/deb
+    mkdir -p $WORKDIR/deb
+    cp $WORKDIR/*.deb $WORKDIR/deb
+    cp $WORKDIR/*.deb $CURDIR/deb
+}
+
+build_tarball(){
+    if [ $TARBALL = 0 ]
+    then
+        echo "Binary tarball will not be created"
+        return;
+    fi
+    get_tar "source_tarball"
+    cd $WORKDIR
+    TARFILE=$(basename $(find . -name 'percona-xtrabackup-*.tar.gz' | sort | tail -n1))
+    if [ "$OS" == "rpm" ]; then
+        if [ "${RHEL}" -eq 8 ]; then
+            export PATH="/usr/local/bin:$PATH"
+        elif [ "${RHEL}" -eq 7 ]; then
+            source /opt/rh/devtoolset-7/enable
+        elif [ "${RHEL}" -eq 6 ]; then
+            source /opt/rh/devtoolset-7/enable
+            source /opt/rh/rh-python36/enable
+        fi
+    fi
```

Comment:
As this code is used in several places it is better to create separate function and just call it when needed

Reply:
Also added CMAKE handling to new function